jlm/hls/backend/rvsdg2rhls/add-forks.cpp

```diff
@@ -36,14 +37,15 @@ add_forks(jlm::rvsdg::region * region)
         add_forks(structnode->subregion(n));
       }
     }
+    // If a node has no inputs it is a constant
+    bool isConstant = node->ninputs() == 0;
     for (size_t i = 0; i < node->noutputs(); ++i)
     {
       auto out = node->output(i);
       if (out->nusers() > 1)
       {
-        std::vector<jlm::rvsdg::input *> users;
-        users.insert(users.begin(), out->begin(), out->end());
-        auto fork = hls::fork_op::create(out->nusers(), *out);
+        std::vector<rvsdg::input *> users(out->begin(), out->end());
+        auto fork = hls::fork_op::create(out->nusers(), *out, isConstant);
         for (size_t j = 0; j < users.size(); j++)
```

Comment:
I just saw it now, but the[ `divert_users()` method of an output](https://github.com/phate/jlm/blob/feb7a6c60627a49bff98b42c5540eabb420a83fa/jlm/rvsdg/node.hpp#L354C1-L361C4) does exactly this.

Reply:
The same a few lines above. The entire add-forks pass could use some refactoring. Merge it for now, it is not part of this PR.

Reply:
divert_users would not work here, since the fork is a user of the same output, and only the other users should be diverted and not it. And the other users are all diverted to separate outputs of the fork, since that's the whole point.

Reply:
Right, it also seems to create a 1-to-1 correspondence between a single user and a fork output.